src/test/java/com/systelab/seed/patient/PatientResourceTest.java

```diff
@@ -10,55 +10,59 @@
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import java.time.LocalDate;
-
 import java.util.List;
+import java.util.UUID;
 
 import static io.restassured.RestAssured.given;
 import static java.util.stream.Collectors.joining;
+import io.restassured.response.Response;
 
 @TmsLink("TC0001_PatientManagement_IntegrationTest")
 @Feature("Patient Test Suite.\n\nGoal:\nThe goal of this TC is to verify that the Patient management actions (CRUD) behave as expected according the specifications and the input values.\n\nEnvironment:\n...\nPreconditions:\nN/A.")
 public class PatientResourceTest extends RESTResourceTest {
 
-    private Patient getPatientData() {
+    private Patient getPatientData(String name, String surname, String email, String medicalNumber, LocalDate dob, String street, String city, String zip) {
         Patient patient = new Patient();
-        patient.setName("Ralph"); // Not NULL
-        patient.setSurname("Burrows"); // Not NULL
-        patient.setEmail("rburrows@gmail.com"); // Format validated
-        patient.setMedicalNumber("123456");
-        patient.setDob(LocalDate.of(1948, 8, 11));
-        Address address = new Address();
-        address.setStreet("E-Street, 90");
-        address.setCity("Barcelona");
-        address.setZip("08021");
-        patient.setAddress(address);
-        return patient;
-    }
-
-    private Patient getPatientData(String name, String surname, String email) {
-        Patient patient = getPatientData();
-        patient.setName(name);
-        patient.setSurname(surname);
-        patient.setEmail(email);
-        return patient;
-    }
 
-    private Patient getPatientData(String name, String surname, String email, String medicalNumber) {
-        Patient patient = getPatientData();
-        patient.setName(name);
-        patient.setSurname(surname);
-        patient.setEmail(email);
-        patient.setMedicalNumber(medicalNumber);
-        return patient;
+        patient.setName(name); //cannot be NULL
+        patient.setSurname(surname); //cannot be NULL
+        if (email != null)
```

Comment:
This if clause is not needed. The email must be set to the parameter value. Please, remove all.